api/envoy/extensions/compression/zstd/compressor/v3/zstd.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+
+package envoy.extensions.compression.zstd.compressor.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.compression.zstd.compressor.v3";
+option java_outer_classname = "ZstdProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/compression/zstd/compressor/v3;compressorv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Zstd Compressor]
+// [#extension: envoy.compression.zstd.compressor]
+
+// [#next-free-field: 6]
+message Zstd {
+  // Reference to http://facebook.github.io/zstd/zstd_manual.html
+  enum Strategy {
+    DEFAULT = 0;
+    FAST = 1;
+    DFAST = 2;
+    GREEDY = 3;
+    LAZY = 4;
+    LAZY2 = 5;
+    BTLAZY2 = 6;
+    BTOPT = 7;
+    BTULTRA = 8;
+    BTULTRA2 = 9;
+  }
+
+  // Set compression parameters according to pre-defined cLevel table.
+  // Note that exact compression parameters are dynamically determined,
+  // depending on both compression level and srcSize (when known).
+  // Default level is ZSTD_CLEVEL_DEFAULT==3.
+  // Special: value 0 means default, which is controlled by ZSTD_CLEVEL_DEFAULT.
+  // Note 1 : it's possible to pass a negative compression level.
+  // Note 2 : setting a level does not automatically set all other compression parameters
+  //   to default. Setting this will however eventually dynamically impact the compression
+  //   parameters which have not been manually set. The manually set
+  //   ones will 'stick'.
+  google.protobuf.UInt32Value compression_level = 1;
```

Comment:
I think zstd is a bit different with other compression algorithm, as [discussion here](https://github.com/facebook/zstd/issues/2012):
> Setting a level does not automatically set all other compression parameters to default. Setting this will however eventually dynamically impact the compression parameters which have not been manually set.
And as [my new test case](https://github.com/envoyproxy/envoy/pull/20434/files#diff-139780ed0cfbc7a354957bbb8b8526d6b60e78bf7ef42dbfbc25f8ec4b0b608cR139), both 0 and 10000 are work for this parameter, so in fact I have no idea how to limit it...